corona-warn-app/cwa-app-android
# Trace Location URL and ID (EXPOSUREAPP-6137, EXPOSUREAPP-6182)

### New
- Generate Trace Location Url [specs](https://github.com/corona-warn-app/cwa-app-tech-spec/blob/6e45edc4df5a8ced091a179bcd40da9bcc162c85/docs/spec/event-registration-client.md#organizer-qr-code-generation)
- Calculate  Trace Location Id  [specs](https://github.com/corona-warn-app/cwa-app-tech-spec/blob/6e45edc4df5a8ced091a179bcd40da9bcc162c85/docs/spec/event-registration-client.md#calculate-tracelocation-id)

### Change
- Updated deep link intent filter to work with the new url requirements 
- Updated QrCode Validation to use `qrCodeDescriptors` from configs and support Base64 and Base32 encoding validation
- Updated Unit Tests and clean them up 

### Testing 
- Test against mock server 
- Create  Trace location 
- Go to test menu -> Event Registration screen
- Scroll to the card `Last Organiser Location`
- You should see location, locationUrl and locationId data displayed for the last created Location 
- If you click on the link the App should respond to and open to ConfirmCheckIn 
- After confirming a CheckIn is added to the list 
- Now go back again to the test menu of the event registration , a card should be displayed for the `Last Attendee Location` (ChecKIn) with the same data above , since Location is the same of organiser and attendee , the data should be exactly the same except the `id` which is Table specific. Data should varies in case you scanned another location which will come next by https://github.com/corona-warn-app/cwa-app-android/pull/2727 .

Please note that url  validation supports both Base32 and Base64 ,but the not the creation which supports only Base64
